### ISSUE TITLE
Appium fails to locate button found in accessibility inspector.

### ISSUE THREAD

**Issue by kevnm67** [appium/appium#10824]
## The problem

- Apple's accessibility inspector finds a custom UIButton Appium fails to locate.
- Expectation: Appium finds the element including its class, label and trait.
- What happened: Appium highlights the button but includes incorrect details in the inspector.

## Environment

* Appium 1.6.1
* Mobile platform/version under test: iOS 11 mainly.  Issue seems OS agnostic.
* Real device or emulator/simulator: Both
* Appium CLI or Appium.app|exe: App

## Details

- Apple's accessibility inspector finds elements Appium is unable to locate or includes incorrect element details.  
- A screenshot captured on an iPad running iOS 11.3.  ![alt text](http://snappyimages.nextwavesrl.netdna-cdn.com/img/e4cd0e5f72efe19ef6c2f7b3b6745a4e.png "iPad running iOS 11.3")
- Apple's accessibility inspector with the correct element details: ![alt text](http://snappyimages.nextwavesrl.netdna-cdn.com/img/8d9280dbbda42ce68a9e619919d686bc.png "Expected details")
- Appium's inspector displaying incorrect element details:  ![alt text](http://snappyimages.nextwavesrl.netdna-cdn.com/img/213554e97f1b7989d7f035becdbbc0d8.png "Incorrect details")

## Link to Appium logs

- https://gist.github.com/kevnm67/04cb6fa96d07e7b2cdea4705f254eaa9

**Comment by mykola-mokhnach** [comment]
I dob't think that anything could be done on Appium side. We get the page source from XCTest/WDA

**Comment by Kitty2014** [comment]
try http://toolsqa.com/mobile-automation/appium/extract-xpath-from-app-using-macaca-inspector/

**Comment by kevnm67** [comment]
Thanks for the response.

@Kitty2014 Ill have QA check it out.  I appreciate the suggestion.

**Comment by mykola-mokhnach** [comment]
Closed as third party issue

**Comment by xshuRoadie** [comment]
Someone opened a ticket reporting a similar issue on WDADriverAgent https://github.com/facebook/WebDriverAgent/issues/854. I am posting the link here so anyone reading this post can track the progress over there. Fingers crossed on a solid solution. We spent quite of time finding a walkaround for our automation due to this issue.
